compiler/parser.nim

```diff
@@ -2401,7 +2401,9 @@ proc parseTopLevelStmt(p: var Parser): PNode =
   while true:
     # nimpretty support here
     if p.tok.indent != 0:
-      if p.firstTok and p.tok.indent < 0: discard
+      if p.firstTok and p.tok.strongSpaceA > 0:
+        parMessage(p, errInvalidIndentation)
+      elif p.firstTok and p.tok.indent < 0: discard
```

Comment:
With this PR, this example below gives `Error: invalid indentation` but shouldn't
```nim
var i: int; echo i
```

Maybe make the first line a special case and check it before the while loop in passes.